### PR TITLE
docs: update documentation

### PR DESCRIPTION
**Pull request by csm-thu**

### CODE REVIEW

File: doc/powerBI.md [hunk 1]
```diff
@@ -1,216 +1,428 @@
...
-- you get the embedded report URL `MyReportURL`
-- the value you need to use for `reportId` key is `MyReportURL?reportId=<reportId>`
-- to get the page name, open your report in PowerBI online, look at the end of your report's URL to get the page name. It should look like `/ReportSection`
+### Use filters in reports
 
-## Report page size recommendation
+#### Static filters
 
-For Scenario View iframe:
+**Static filters** let you fiter a given field on a hard-coded value. You simply need to define an object with the **table** and **column** to filter, and the **filter value** to apply. The keys of this object are:
```
Comment (esasova) [inline doc/powerBI.md:353]: typo: fiLter :slightly_smiling_face:

File: doc/powerBI.md [hunk 2]
```diff
@@ -1,216 +1,428 @@
...
+| ---------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `title`          | report title, per language, to display in the let menu in the Dashboards view (it is not used for reports in Scenario view)                                                      |
+| `reportId`       | id of the report; it can be found in PowerBI, in the URL of your report: `myReportURL?reportId=<reportId>`                                                                       |
+| `settings`       | an object to configure the [PowerBI report settings](https://github.com/microsoft/powerbi-models/blob/0d326572c4253fd9f89b73a0d8df1ae46318a860/src/models.ts#L1070)              |
+| `staticFilters`  | an array of static filters objects (see section [Static filters](#how-to-use-static-filters))                                                                                    |
```
Comment (esasova) [inline doc/powerBI.md:83]: the link opens just PowerBI.md file, not the right section given by anchor

File: doc/powerBI.md [hunk 3]
```diff
@@ -1,216 +1,428 @@
...
+| `dynamicFilters` | an array of dynamic filters objects (see section [Dynamic filters](#how-to-use-filters-on-the-current-scenario-dynamic-filter))                                                  |
```
Comment (esasova) [inline doc/powerBI.md:84]: the same

File: doc/scenarioParametersConfiguration.md [hunk 4]
```diff
@@ -44,220 +33,165 @@ parameters to display for each one. A tab is generated for each parameter group,
...
+model.
 
 ### Parameters of type enum
 
-When using a scenario parameter of type enum, you have to declare in the webapp configuration the list of possible
-values the enum can take, and the labels that will displayed in the front-end interface for each value. This can be done
-in the `PARAMETERS` constant of the scenario parameters configuration, by adding the field `enumValues` in the object
-representing your parameter, where `enmValues` is a list of objects with two properties:
+When using a scenario parameter of type enum, you have to declare in your solution description the list of possible
+values the enum can take (values sent to and received from the Cosmo Tech API), and the labels that will displayed in
```
Comment (esasova) [inline doc/scenarioParametersConfiguration.md:83]: typo: will BE displayed :slightly_smiling_face:

File: doc/scenarioParametersConfiguration.md [hunk 5]
```diff
@@ -44,220 +33,165 @@ parameters to display for each one. A tab is generated for each parameter group,
...
+| -------------------------------------------- | ------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `id`                                         | mandatory          | unique id of the parameter group                                                                                                                         |
+| `parameters`                                 | mandatory          | fields to list the ids of the parameters that are part of this group                                                                                     |
+| `labels`                                     | optional           | a dict of labels for this parameter group, the keys of the dict being the code ISO 639-1 of the language,                                                |
+| and the values being the corresponding label |
+| `authorizedRoles`                            | optional           | an array of [APP_ROLES](../src/config/Profiles.js) allowed to see the scenario parameters tab (by default, all tabs are visible)                         |
```
Comment (esasova) [inline doc/scenarioParametersConfiguration.md:60]: _Profiles.js_ doesn't exist anymore (and is it possible to define roles in web app ?) :thinking:

File: doc/scenarioParametersCustomization.md [hunk 6]
```diff
@@ -4,149 +4,134 @@ This document describes advanced methods to customize the scenario parameters vi
...
+        - key: 'BARREL'
+          value: 'bl'
+        - key: 'CUBIC_METRE'
+          value: 'm³'
+      subType: 'RADIO'
 ```
 
-If the _varType_ of the parameter above is `%DATASETID%`, then the extended _varType_ for this parameter will be
-`%DATASETID%-TABLE`.
+This sub-type will be concatenated after the `varType` value. For instance, if the `varType` of a parameter is
```
Comment (esasova) [inline doc/scenarioParametersCustomization.md:123]: typo: sybtype (lines 123 and 124) :slightly_smiling_face:

File: doc/scenarioParametersCustomization.md [hunk 7]
```diff
@@ -4,149 +4,134 @@ This document describes advanced methods to customize the scenario parameters vi
...
+`%DATASETID%` and its sub-type is `TABLE`, then the extended `varType` for this parameter will be `%DATASETID%-TABLE`.
 
-You will then be able to use this extended _varType_ in the four mapping files (see section "Define a custom
-_varType_"). For any of those files, if the extended _varType_ of a parameter is not found in the mapping dict, the
-webapp will fallback on its _varType_. This means that if you want to keep the same behavior for an extended _varType_
-as for the orgiinal _varType_, you can omit to declare the extended varType in the mapping dicts.
+You will then be able to use this extended `varType` in the four mapping files (see section "Define a custom
+`varType`"). For any of those files, if the extended `varType` of a parameter is not found in the mapping dict, the
+webapp will fallback on its `varType`. This means that if you want to keep the same behavior for an extended `varType`
```
Comment (esasova) [inline doc/scenarioParametersCustomization.md:128]: I have just learned this rule: **fallback** is a noun  and **fall back** is a verb :rofl:

**Review by esasova — APPROVED** [review]
Just some typos, LGTM :+1:

File: doc/migrationGuides/v5.0.0.md [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+
+#### New settings in Solution and Workspace data
+
+Because of the new **workspace selector** feature, we must **provide all the information** required by the webapp in Solution and Workspace objects stored by the Cosmo Tech API, and be able to **override** them for local webapp development. **The following files have thus been removed**:
+
+- _src/config/InstanceVisualization.js_ → Workspace data
+- _src/config/PowerBI.js_ → Workspace data
+- _src/config/ScenarioParameters.js_ → Solution data
+
+Optionally, **you can now override the data of multiple solutions and workspaces** by using the files **src/config/overrides/Solutions.js** and _src/config/overrides/Workspaces.js_. Complete examples of format for these files can be found in the GitHub repository:
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply doc/migrationGuides/v5.0.0.md:89]: Probably a formatting error by the keyboard operator :grin:

File: doc/scenarioParametersConfiguration.md (hunk 5 above)
Comment (csm-thu) [reply doc/scenarioParametersConfiguration.md:60]: Fixed the link to point to the back-end documentation with the up-to-date list of roles :pray: 
These roles can still be used in `authorizedRoles` in the parameters groups configuration, but we cannot define other roles than the existing ones.

Comment (esasova) [reply doc/scenarioParametersConfiguration.md:60]: thank you ! :pray:

**Review by ardvi — APPROVED** [review]
LGTM :+1:

**Review by nborde-CSM — APPROVED** [review]
Only one remark but LGTM :+1:

File: doc/applicationInsights.md [hunk 9]
```diff
@@ -1,51 +1,75 @@
...
-- in the Azure portal:
-    - open the page of your Static Web Apps instance
-    - select "Application Insights" in the left panel
-    - select "Yes" next to "Enable Application Insights" (see note below if you can't enable Application Insights)
-    - click on "Save"
-- then, edit the configuration file [src/config/ApplicationInsights.js](../src/config/ApplicationInsights.js) in the source code of your webapp:
-    - replace the value of **APPLICATION_INSIGHTS_INSTRUMENTATION_KEY** with the instrumentation key that has been
-      generated for your application
-    - set **ENABLE_APPLICATION_INSIGHTS** to true
+Our application uses _Application Insights_, an Azure service, to monitor some user actions and webapp performance.
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply doc/applicationInsights.md:3]: I'm not sure, it seems both are valid, but the singular form is more commonly used ?

**Pull request merged**